File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.swift
import Foundation

func solution(_ number:Int, _ limit:Int, _ power:Int) -> Int {
    let list = [Int](1 ... number)
    var result = 0
    
    // i = 1,2,3,4,5,6,7,8,9,10
    for i in list {
        var cnt = 0
        for j in 1 ... Int(sqrt(Double(i))) {
            if i % j == 0 {
                if j*j == i {
                    cnt += 1
                }else{
                    cnt += 2
                }
            }
        }
        result += cnt <= limit ? cnt : power
    }
    

    return result
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let result = 0
    let list = []
    
    for(let i=1; i <= number; i++) {
        let cnt = 0
        const max = Number((Math.sqrt(i)))
        
        for(let j=1; j<=max; j++) {
            if(i%j === 0) {
                if(j*j === i) { cnt += 1 } 
                else { cnt += 2 }
            }        
        }
        list.push(cnt)
    }
    
    for(let i of list) {
        if(i > limit) { result+= power}
        else { result += i}
    }
    
    return result
}
